deploy-image/src/index.ts

```diff
@@ -1,179 +0,0 @@
-import { context, getOctokit } from '@actions/github'
-import { promises as fs } from 'fs'
-import { getBooleanInput } from '../../lib/core'
-import {
-  getEventData,
-  getEventName,
-  getRepository,
-  getToken,
-  isEvent,
-} from '../../lib/github'
-import {
-  getComponentVersion,
-  getDeployType,
-  getImageLabels,
-  getImagePrefixAndTags,
-  GHCR,
-} from './metadata'
-
-const METADATA_FILENAME = 'metadata.json'
-
-async function run() {
-  const { component: defaultComponent } = getRepository()
-  const component = getInput('component') || defaultComponent
-  const dockerUser = getInput('docker-hub-user', { required: true })
-  const dockerPassword = getInput('docker-hub-password', { required: true })
-  const ghcrUser = getInput('ghcr-user') || context.actor
-  const ghcrPassword = getInput('ghcr-password') || getToken()
-  const dryRun = getBooleanInput('dry-run', false)
-  const dockerfile = getInput('dockerfile') || './Dockerfile'
-  const eventName = getEventName(['push', 'delete'])
-  const eventData = await getEventData<typeof eventName>()
-
-  const type = getDeployType()
-  const { prefix, tags } = await getImagePrefixAndTags()
-  info(`Image deploy type: ${type}`)
-  info(`Image prefix: ${prefix ?? 'none'}`)
-  info(`Image name: exivity/${component}`)
-  info(`Image tags: ${tags.join(',')}`)
-
-  // This is hard to test so we allow emulating this event with an env var
-  if (
-    isEvent(eventName, 'delete', eventData) ||
-    !!process.env['EMULATE_DELETE']
-  ) {
-    if (type !== 'branch') {
-      info(`Not deleting images for deploy type "${type}"`)
-      return
-    }
-
-    const ghToken = getToken()
-    const octokit = getOctokit(ghToken)
-
-    const { data: versions } =
-      await octokit.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
-        org: 'exivity',
-        package_type: 'container',
-        package_name: component,
-        per_page: 100,
-      })
-
-    if (versions.length === 100) {
-      warning(
-        `Please clean up package versions or implement pagination, we found 100 or more!`
-      )
-    }
-
-    if (!versions.length) {
-      info('No package versions found')
-    }
-
-    // Look for versions with the same tags as the ones we'll delete
-    for (const version of versions) {
-      const tagOverlap = tags.filter((tag) =>
-        version.metadata?.container?.tags?.includes(tag)
-      )
-
-      if (tagOverlap.length > 0) {
-        info(
-          `Deleting package version ${
-            version.id
-          }, it was tagged with "${version.metadata?.container?.tags?.join(
-            ','
-          )}"`
-        )
-        await octokit.rest.packages.deletePackageVersionForOrg({
-          org: 'exivity',
-          package_type: 'container',
-          package_name: component,
-          package_version_id: version.id,
-        })
-      } else {
-        warning('Could not find matching package version to delete')
-      }
-    }
-
-    return
-  }
```

Comment:
We'd need to extract this logic into the new action and document this. I can take on this part if you want

Reply:
we discussed this, I think we can remember it's somewhere in the git tree and more on.